python/pyiceberg/table/snapshots.py

```diff
@@ -117,12 +125,369 @@ def manifests(self, io: FileIO) -> List[ManifestFile]:
             return list(read_manifest_list(file))
         return []
 
+    def added_data_files(self, io: FileIO) -> Generator[DataFile, None, None]:
+        for manifest in self.manifests(io):
+            yield from [entry.data_file for entry in manifest.fetch_manifest_entry(io)]
```

Comment:
These entries are not necessarily live. For live data files, check `entry.status < DELETED` (2)